main/src/library/MutList.flix

```diff
@@ -653,17 +651,10 @@ namespace MutList {
     ///
     pub def reserve!(n: Int32, v: mut MutList[a]): Unit & Impure =
         let MutList(a, l) = v;
-        let len = deref l;
-        if (len == 0)
-            a := Array.new(default, n)
-        else {
-            let f = i ->
-                if (i < len)
-                    (deref a)[i]
-                else
-                    default;
-            a := Array.init(f, len + n)
-        }
+        let empty = capacity(v) - deref l;
+        if (n > empty)
+            a := Array.append(deref a, [default; n - empty])
+        else ()
```

Comment:
`reserve!` is a key operation, but this implementation here is slower than it could be. (It allocates two arrays, instead of just one). It would be better to allocate an array of the new full length and then copy the elements into it.